hepcrawl/loaders.py

```diff
@@ -165,5 +164,40 @@ class HEPLoader(ItemLoader):
     )
     urls_out = ListToValueDict()
 
+    def get_dois(self, dois=None, material=None):
+        """Adds `dois` field to `self`.
+
+        Args:
+            dois (list):The `dois` represented by a list of strings.
+            material (str):The `material` of the dois.
+
+        Examples:
+            Using `dois`::
+
+                >>> from hepcrawl.items import HEPRecord
+                >>> record = HEPLoader(item=HEPRecord(), selector=selector)
+                >>> record.get_dois(dois=dois_list_str)
+
+            Using `dois` and `material`::
+
+                >>> from hepcrawl.items import HEPRecord
+                >>> record = HEPLoader(item=HEPRecord(), selector=selector)
+                >>> record.get_dois(dois=dois_list_str, material='preprint')
+        """
+        if dois is None or []:
+            return
+
+        doi_list = []
+        for _doi in dois:
```

Comment:
You could also rename the `dois` parameter there to `dois_values`, then you can use `dois` instead of `doi_list`:
```
def get_dois(self, dois_values=None, material=None):
    ...
    dois = []
    for doi_value in dois_values:
        doi = {...}
        dois.append(doi)
```

though a more pythonic way might be using an internal function:
```python
def _build_doi(doi_value, material):
    doi = {'value': doi_value}
    if material:
        doi['material'] = material

    return doi

dois = [
    _build_doi(doi_value, material)
    for doi_value in dois_values
]
```